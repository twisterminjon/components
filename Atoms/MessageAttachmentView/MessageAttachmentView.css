.messageattachmentview {
  height: 100%;

  display: flex;
  justify-content: center;
  align-items: center;

  padding: 16px;

  flex: 1;
}

.messageattachmentview img {
  position: relative;
  height: 95%;
  object-fit: contain;
}

.messageattachmentview img:before {
  content: ' ';
  display: block;

  border-radius: 5px;
  background: repeating-linear-gradient(
    45deg,
    var(--messagepage__received_bg),
    var(--messagepage__received_bg) 10px,
    var(--messagepage__sent_bg) 10px,
    var(--messagepage__sent_bg) 20px
  );

  position: absolute;
  top: -10px;
  left: 0;
  height: 100%;
  width: 100%;
}

.messageattachmentview img:after {
  content: 'ðŸ’” Image not found';
  display: block;

  font-size: 16px;
  text-align: center;
  padding: 10px 0px;
  color: var(--messagepage__received_fg);

  position: absolute;
  top: 5px;
  left: 0;
  width: 100%;
}
