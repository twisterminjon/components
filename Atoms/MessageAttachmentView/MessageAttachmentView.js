import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './MessageAttachmentView.css'

export default class MessageAttachmentView extends Component {
  static propTypes = {
    /** Content for the attachment */
    children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
    /** Called after the attachment is clicked */
    onClick: PropTypes.func,
    /** ClassName for the wrapper */
    className: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
  }
  static defaultProps = {
    onClick: () => {},
    className: '',
    style: {},
  }

  render() {
    const { onClick, children, className, style, ...rest } = this.props

    return (
      <div onClick={onClick} className={`messageattachmentview ${className}`.trim()} style={style} {...rest}>
        {children}
      </div>
    )
  }
}
