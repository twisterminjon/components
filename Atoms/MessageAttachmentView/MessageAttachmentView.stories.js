import React from 'react'

import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs'

import MessageAttachmentView from './MessageAttachmentView'

const stories = storiesOf('Provider/v1/Atoms', module)
stories.addDecorator(withKnobs)

stories.add('MessageAttachmentView', () => {
  return (
    <div
      style={{
        width: 375,
        height: 628,
      }}>
      <MessageAttachmentView
        onClick={() => {
          alert('clicked')
        }}>
        <img alt="attachment" src="https://www.fillmurray.com/800/1200" />
      </MessageAttachmentView>
    </div>
  )
})
