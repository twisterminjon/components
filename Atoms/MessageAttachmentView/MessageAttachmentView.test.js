import React from 'react'
import MessageAttachmentView from './MessageAttachmentView'

describe('MessageAttachmentView', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(
      <MessageAttachmentView onClick={mockFun}>
        <img alt="attachment" src="https://www.fillmurray.com/800/1200" />
      </MessageAttachmentView>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
