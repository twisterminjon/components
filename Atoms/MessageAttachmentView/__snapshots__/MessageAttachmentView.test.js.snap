// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageAttachmentView matches the snapshot 1`] = `
<div
  className="messageattachmentview"
  onClick={[MockFunction]}
  style={Object {}}
>
  <img
    alt="attachment"
    src="https://www.fillmurray.com/800/1200"
  />
</div>
`;
